app/graphql/mutations/create_asset.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class Mutations::CreateAsset < Mutations::BaseMutation
+  argument :signed_blob_id, String, "Signed blob ID generated via `createDirectUpload` mutation", required: true
+  argument :parent_id, ID, required: true
+
+  field :asset, Types::DigitalObject::AssetType, null: true
+
+  def resolve(signed_blob_id:, parent_id:)
+    parent = DigitalObject::Item.find(parent_id)
+    ability.authorize! :create_objects, parent.primary_project
+
+    blob = ActiveStorage::Blob.find_signed(signed_blob_id)
+
+    asset = initialize_child_asset(parent, blob.filename.to_s)
+
+    begin
+      asset.resources['master'] = initialize_asset_resource_for_blob(asset, blob, Hyacinth::Config.resource_storage)
```

Comment:
It might be good to use a constant here, and below, instead of literal string `'master'`.

Reply:
if we were to use a constant, does it make more sense for it to be at this layer, or on the model? If the latter, I would advocate taking up separately, maybe in a `.default_resource &block` type implementation that completely removes the reference from the mutation layer.

Reply:
"taking up separately" subsequent to this PR, that is

Reply:
Definitely in the model, and subsequent PR sounds good!